### PR TITLE
Some potential serialization issues resolved

### PR DESCRIPTION
**Pull request by dmmiller612**
I noticed that there are a couple of scenarios where serialization issues can arrise. This should handle that, by using the native multilayernetwork.

### CODE REVIEW

**Comment by dmmiller612** [comment]
I may add the sparknetwork as well. However, I haven't seen it have any issues with transformations, since it extends a different spark class.

File: deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/AutoEncoderWrapper.scala [hunk 1]
```diff
@@ -48,7 +48,7 @@ abstract class AutoEncoderWrapper[E <: AutoEncoderWrapper[E, M], M <: AutoEncode
         val set2 = dataset.select($(inputCol)).rdd.map(mapVectorFunc)
         val ds = set2.map(v => new DataSet(Nd4j.create(v.toArray), Nd4j.create(v.toArray)))
         val fitted = sparkNet.fit(ds)
-        sparkNet
+        fitted.clone()
     }
```
Comment (agibsonccc) [inline deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/AutoEncoderWrapper.scala]: Doesn't this get expensive? The averaged network isn't used anywhere else. Do we really need to clone?

File: deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/spark-2/scala/org/deeplearning4j/spark/ml/impl/AutoEncoder.scala [hunk 2]
```diff
@@ -38,10 +39,10 @@ class AutoEncoder(uid: String) extends AutoEncoderWrapper[AutoEncoder, AutoEncod
...
 
-class AutoEncoderModel(uid: String, sparkDl4jMultiLayer: SparkDl4jMultiLayer) extends AutoEncoderModelWrapper[AutoEncoderModel](uid, sparkDl4jMultiLayer) {
+class AutoEncoderModel(uid: String, multiLayerNetwork: MultiLayerNetwork) extends AutoEncoderModelWrapper[AutoEncoderModel](uid, multiLayerNetwork) {
 
     override def udfTransformer = udf[Vector, Vector](vec => {
-        val out = sparkDl4jMultiLayer.getNetwork.feedForwardToLayer($(compressedLayer), Nd4j.create(vec.toArray))
+        val out = multiLayerNetwork.feedForwardToLayer($(compressedLayer), Nd4j.create(vec.toArray))
         val mainLayer = out.get(out.size() - 1)
         val size = mainLayer.size(1)
         println(size)
```
Comment (agibsonccc) [inline deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/spark-2/scala/org/deeplearning4j/spark/ml/impl/AutoEncoder.scala]: Get rid of the println here?

File: deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/AutoEncoderWrapper.scala (hunk 1 above)
Comment (dmmiller612) [inline deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/AutoEncoderWrapper.scala]: yeah, I get a little zealous with the serializable thing. I have a fix for it now.

File: deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/spark-2/scala/org/deeplearning4j/spark/ml/impl/AutoEncoder.scala (hunk 2 above)
Comment (dmmiller612) [inline deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/spark-2/scala/org/deeplearning4j/spark/ml/impl/AutoEncoder.scala]: yep, fixed

**Pull request merged**